Terminal.Gui/View/Layout/Pos.cs

```diff
@@ -0,0 +1,694 @@
+    ///     internally by the layout system to determine where a View should be positioned.
+    /// </summary>
+    /// <param name="width">The width of the area where the View is being positioned (Superview.ContentSize).</param>
+    /// <returns>
+    ///     An integer representing the calculated position. The way this position is calculated depends on the specific
+    ///     subclass of Pos that is used. For example, PosAbsolute returns a fixed position, PosAnchorEnd returns a
+    ///     position that is anchored to the end of the layout, and so on.
+    /// </returns>
+    internal virtual int Anchor (int width) { return 0; }
+
+    /// <summary>
+    ///     Calculates and returns the position of a <see cref="View"/> object. It takes into account the dimension of the
+    ///     superview and the dimension of the view itself.
+    /// </summary>
+    /// <param name="superviewDimension">
+    ///     The dimension of the superview. This could be the width for x-coordinate calculation or the
+    ///     height for y-coordinate calculation.
+    /// </param>
+    /// <param name="dim">The dimension of the View. It could be the current width or height.</param>
+    /// <param name="us">The View that holds this Pos object.</param>
+    /// <param name="dimension">Width or Height</param>
+    /// <returns>
+    ///     The calculated position of the View. The way this position is calculated depends on the specific subclass of Pos
+    ///     that
+    ///     is used.
+    /// </returns>
+    internal virtual int Calculate (int superviewDimension, Dim dim, View us, Dimension dimension) { return Anchor (superviewDimension); }
+
+    /// <summary>
+    ///     Diagnostics API to determine if this Pos object references other views.
+    /// </summary>
+    /// <returns></returns>
+    internal virtual bool ReferencesOtherViews () { return false; }
+}
+
+/// <summary>
+///    Represents an absolute position in the layout. This is used to specify a fixed position in the layout.
+/// </summary>
+/// <remarks>
+///     <para>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Pos"/> class to create <see cref="Pos"/> objects instead.
+///     </para>
+/// </remarks>
+/// <param name="position"></param>
+public class PosAbsolute (int position) : Pos
+{
+    /// <summary>
+    ///    The position of the <see cref="View"/> in the layout.
+    /// </summary>
+    public int Position { get; } = position;
+
+    /// <inheritdoc />
+    public override bool Equals (object other) { return other is PosAbsolute abs && abs.Position == Position; }
+
+    /// <inheritdoc />
+    public override int GetHashCode () { return Position.GetHashCode (); }
+
+    /// <inheritdoc />
+    public override string ToString () { return $"Absolute({Position})"; }
+
+    internal override int Anchor (int width) { return Position; }
```

Comment:
What? 🤔

Reply:
`Anchor` *Is* confusing, if that's what you're asking about.  I don't have a great answer.

Reply:
Well, it's all part of the same issue, really.

(Specifically, though, the "What?" is about Anchor returning Position, because Position is...probably not a good name for that.)

Zoomed out a bit, though:

Pos and Dim are sorta like a higher-level incarnation of the _components_ of a Point and Size, respectively, but with behavioral preference along for the ride.

Most of what they do is really/should be the responsibility of Views or IPositionable or _whatever_ themselves. Mixing data and behavior in what is conceptually a primitive is discouraged, partially because it leads to exactly this sort of confusion - especially since the type is likely to be used in places that stretch the meaning of the names of properties even further.

With the _behavior_ defined more generically, the need for actual separate types evaporates, and a simple enum is then sufficient.

For a common example, that's how HTML/CSS do it. The enum is `position` (unfortunate name in my opinion) and it is an enum for the desired positioning style. Then, the relevant properties on the DOM element itself are used and the rules of the selected behavior applied.

https://www.w3schools.com/Css/css_positioning.asp